deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/BaseDeepLinkDelegate.java

```diff
@@ -210,4 +236,47 @@ private static void notifyListener(Context context, boolean isError, Uri uri,
   public boolean supportsUri(String uriString) {
     return findEntry(uriString) != null;
   }
+
+  /**
+   * Validate a user's configuration of pathVariableReplacementValue.
+   */
+  private void validatePathVariableReplacementValue(String pathVariableReplacementValue) {
+    if (pathVariableReplacementValue.matches("[a-zA-Z0-9/-]*"))
+      throw new RuntimeException("Only a-z, A-Z, 0-9, and - are allowed in a "
+        + "pathVariableReplacementValue. Currently it is: " + pathVariableReplacementValue);
+  }
+
+  /**
+   * Ensure that every key-to-be-replaced declared by all registries have a corresponding key in
+   * the user's injected mapping of pathVariableReplacements. If not, throw an exception and tell
+   * the user which keys aren't present.
+   * @param registries
+   * @param pathVariableReplacements
+   */
+  private void validatePathVariableReplacements(List<? extends BaseRegistry> registries, Map<String,
+    String> pathVariableReplacements) {
+    HashSet<String> keysUnion = new HashSet<>();
```

Comment:
Is this pretty fast? We could do it in a background thead if slow.

Also I feel like we could write this a lot more clearly in kotlin. Seems like it could just be an internal util instead of a member of this class, and then write that in Kotlin?

Reply:
I'm imagining a pretty small number of keys to union (like 1 per DLD instance), but we can totally do it on a background, esp since it will run at every DLD init.

Reply:
Bump on considering:
1) background thread
2) moving to kotlin extension
3) instead of (1) base delegate could take a debug flag to enable / disable this validation